### ISSUE TITLE
Horizontal line/Spacer renders before image

### ISSUE THREAD

**Issue by kokoISnoTarget** [Inlyne-Project/inlyne#305]
```
![](https://placehold.co/200x100/EEE/31343C)

<hr>
```
Renders in Inlyne (4.2): 
![Screenshot_2024-04-07-11-33-02_5120x1263](https://github.com/Inlyne-Project/inlyne/assets/72217393/5b7412ac-c967-4287-9e32-064d28604243)

GitHub:
![](https://placehold.co/200x100/EEE/31343C)

<hr>

(edit): This does not occur when align center or right is used, so the problem lies somewhere in how we handle image rows because https://github.com/Inlyne-Project/inlyne/blob/32619fff5ab60e1217d12be43b91e679042eb0f2/src/interpreter/mod.rs#L341C9-L352C10

**Comment by CosmicHorrorDev** [comment]
Yeah the code for handling inline images has been notoriously buggy. I'll work on adding some more tests on how that part of the code should behave over time since there are a lot of different variations that people use for their README's headers
